### PR TITLE
Update GQL to expose StaleStatus and StaleStatusCause

### PR DESCRIPTION
**Pull request by smackesey**
### Summary & Motivation

This updates GQL to expose the new StaleStatus and StaleStatusCause data.

### How I Tested These Changes

Unit tests (in progress)

### CODE REVIEW

**Comment by smackesey** [comment]
Current dependencies on/for this PR:
* master
  * **PR #11950** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/11950" target="_blank">Graphite</a> 
    * **PR #11951** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/11951" target="_blank">Graphite</a> 
      * **PR #11952** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/11952" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/dagster-io/dagster/11952?utm_source=stack-comment).

File: python_modules/dagster-graphql/dagster_graphql/schema/asset_graph.py [hunk 1]
```diff
@@ -327,6 +348,14 @@ def is_multipartitioned(self) -> bool:
...
         )
 
+    def convert_stale_status(self, status: StaleStatus) -> GrapheneAssetStaleStatus:
+        if status == StaleStatus.FRESH:
+            return GrapheneAssetStaleStatus.FRESH  # type: ignore  # (bad stubs)
+        elif status == StaleStatus.STALE:
+            return GrapheneAssetStaleStatus.STALE  # type: ignore  # (bad stubs)
+        else:
+            return GrapheneAssetStaleStatus.UNKNOWN  # type: ignore  # (bad stubs)
+
```
Comment (bengotow) [inline python_modules/dagster-graphql/dagster_graphql/schema/asset_graph.py:358]: I'd defer to @sryza but I think there's some sort of magic incantation you can do to have this conversion done automatically, maybe renaming `GrapheneAssetStaleStatus` to `GrapheneStaleStatus` so it matches the python enum name would do it?

**Review by bengotow — APPROVED** [review]
This looks great to me!

File: python_modules/dagster-graphql/dagster_graphql/schema/asset_graph.py (hunk 1 above)
Comment (smackesey) [reply python_modules/dagster-graphql/dagster_graphql/schema/asset_graph.py:358]: I was wondering about this...

Comment (sryza) [reply python_modules/dagster-graphql/dagster_graphql/schema/asset_graph.py:358]: I think there may be, but I don't know what it is. @prha or @alangenfeld or @clairelin135 might?

Comment (alangenfeld) [reply python_modules/dagster-graphql/dagster_graphql/schema/asset_graph.py:358]: can try `from_enum` https://docs.graphene-python.org/en/latest/types/enums/

File: python_modules/dagster-graphql/dagster_graphql/schema/asset_graph.py [hunk 2]
```diff
@@ -542,7 +556,23 @@ def resolve_configField(self, _graphene_info: ResolveInfo) -> Optional[GrapheneC
...
         return self._external_asset_node.compute_kind
 
-    def resolve_currentLogicalVersion(self, _graphene_info: ResolveInfo) -> Optional[str]:
+    def resolve_staleStatus(self, graphene_info: ResolveInfo) -> Any:  # (GrapheneAssetStaleStatus)
+        return self.stale_status_loader.get_status(self._external_asset_node.asset_key)
+
+    def resolve_staleStatusCauses(
+        self, graphene_info: ResolveInfo
+    ) -> Sequence[GrapheneAssetStaleStatusCause]:
+        lineage = self.stale_status_loader.get_status_causes(self._external_asset_node.asset_key)
```
Comment (sryza) [inline python_modules/dagster-graphql/dagster_graphql/schema/asset_graph.py:565]: nitpick: lineage -> causes

**Review by sryza — APPROVED** [review]

**Pull request merged**